services/venice-router/src/main/java/com/linkedin/venice/router/MetaDataHandler.java

```diff
@@ -477,6 +504,69 @@ private void handleCurrentVersionLookup(ChannelHandlerContext ctx, VenicePathPar
     setupResponseAndFlush(OK, OBJECT_MAPPER.writeValueAsBytes(response), true, ctx);
   }
 
+  /**
+   * Handles the discovery of blob transfer nodes based on store settings.
+   * Retrieves host names for live DVC nodes ready to transfer blobs.
+   * Only returns host names from nodes that have completed a full push and are active.
+   *
+   * @return a response with a list of host names for live DVC nodes; returns an empty list if no live nodes are found or if conditions are not met
+   */
+  private void handleBlobDiscovery(ChannelHandlerContext ctx, VenicePathParserHelper helper, HttpRequest request)
+      throws IOException {
+
+    // i.e. /blob_discovery?store_name=storeName&store_version=22&store_partition=2
+    Map<String, String> queryParams = helper.extractQueryParameters(request);
+    String storeName = queryParams.getOrDefault(NAME, "");
+    String storeVersion = queryParams.getOrDefault(STORE_VERSION, "");
+    String storePartition = queryParams.getOrDefault(STORE_PARTITION, "");
+
+    if (StringUtils.isEmpty(storeName) || StringUtils.isEmpty(storeVersion) || StringUtils.isEmpty(storePartition)) {
+      byte[] errBody =
+          (String.format(REQUEST_BLOB_DISCOVERY_MISSING_QUERY_PARAMS, storeName, storeVersion, storePartition))
+              .getBytes();
+      setupResponseAndFlush(BAD_REQUEST, errBody, false, ctx);
+      return;
+    }
+
+    Store store = storeRepository.getStore(storeName);
+    if (store == null) {
+      byte[] errBody = (String.format(REQUEST_ERROR_STORE_NOT_FOUND_IN_CLUSTER, storeName, clusterName)).getBytes();
+      setupResponseAndFlush(NOT_FOUND, errBody, false, ctx);
+      return;
+    }
+
+    if (!store.isBlobTransferEnabled() || store.isHybrid()) {
+      byte[] errBody = (String.format(REQUEST_BLOB_DISCOVERY_ERROR_INVALID_SETTINGS, storeName)).getBytes();
+      setupResponseAndFlush(FORBIDDEN, errBody, false, ctx);
+      return;
+    }
+
+    BlobDiscoveryResponse response = new BlobDiscoveryResponse();
+    try {
+      // gets the instances for a FULL_PUSH for the store's version and partitionId
+      // gets the instance's hostnames from its keys & filter to include only live instances
+      Map<CharSequence, Integer> instances = pushStatusStoreReader.getPartitionStatus(
```

Comment:
as[ per this thread](https://linkedin-randd.slack.com/archives/C06HAFXUMPZ/p1716926655209899), we will revisit this after the changes have been made in future.

Reply:
That's right, let's not post links to LinkedIn's internal slack. That being said, this may be a good opportunity to remind that we have the open source slack available to us, and it may be preferable to move all conversations about this project to a new public channel in the open source slack (: